fenix/app/src/main/java/org/mozilla/fenix/settings/advanced/LocaleUtils.kt

```diff
@@ -0,0 +1,232 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.advanced
+
+import java.util.Locale
+
+/**
+ * Helper class for displaying Locale name.
+ */
+object LocaleUtils {
+
+    /**
+     * Returns a name for the locale that is appropriate for display to the user.
+     */
+    fun getDisplayName(locale: Locale): String {
+        val displayName = locale.getDisplayName(locale)
+            .replaceFirstChar { if (it.isLowerCase()) it.titlecase(locale) else it.toString() }
+        if (displayName.equals(locale.toString(), ignoreCase = true)) {
+            return LOCALE_TO_DISPLAY_NATIVE_NAME_MAP[locale.toString()] ?: displayName
+        }
+        return displayName
+    }
+
+    private val LOCALE_TO_DISPLAY_NATIVE_NAME_MAP: Map<String, String> = mapOf(
```

Comment:
I'm assuming there is no automation or anything that expected these values to be where they were originally. Maybe @Delphine could confirm it is okay to move around?